### ISSUE TITLE
OwnedStruct doesn't preserve field order

### ISSUE THREAD

**Issue by marcbowes** [amazon-ion/ion-rust#200]
```
pub struct OwnedStruct {
    text_fields: HashMap<Rc<str>, Vec<(OwnedSymbolToken, OwnedElement)>>,
```

So, given 

```
{
  foo:"bar",
  baz:"zor",
}
```

the output of a program that iterates over the struct and prints each field name would not preserve the "foo", "baz" ordering (what I expect) or even be deterministic (HashMap seeds the hasher each time).

**Comment by almann** [comment]
This probably should be configurable--technically order is not guaranteed per spec (Ion and JSON), but we could support this with a data structure that maintained the insertion order and I have seen a similar use case in other places.

**Comment by marcbowes** [comment]
Just to add a bit more color.. This is from the qldb shell:

```
qldb> select * from information_schema.user_tables
{
  tableId: "7NmEOVTm9XCClLRhkuAzw0",
  name: "bar",
  indexes: [
  ],
  status: "ACTIVE"
},
{
  tableId: "1nmeTZw7uhuI2kBXrbYh8x",
  name: "foo",
  indexes: [
    {
      expr: "[bar]",
      indexId: "4zJJ3hSDASCJVv4PmuJXmH",
      status: "ONLINE"
    }
  ],
  status: "ACTIVE"
}
2 documents in bag (read-ios: 2, server-time: 11ms, total-time: 561ms)
```

and now with `--format table`

```
qldb> select * from information_schema.user_tables
+------------------------+--------+------+------------------------+
| indexes                | status | name | tableId                |
+=================================================================+
| []                     | ACTIVE | bar  | 7NmEOVTm9XCClLRhkuAzw0 |
|------------------------+--------+------+------------------------|
| [struct with 3 fields] | ACTIVE | foo  | 1nmeTZw7uhuI2kBXrbYh8x |
+------------------------+--------+------+------------------------+
2 documents in bag (read-ios: 2, server-time: 10ms, total-time: 233ms)
```

Note the order of the fields in the Ion text output has changed compared to the table view. (and, also, that I haven't rendered the sub-struct properly.)

**Comment by zslayton** [comment]
Fixed by #429.
